### DISCUSSION TITLE
Is there information on broadcasting frequency for different devices?

### DISCUSSION THREAD

**Discussion by hurricup** [custom-components/ble_monitor#563] [Q&A] [answered]
This is unrelated to this particular component, but more like BLE in general.
I've got few thermometers, but they are broadcasting like once per 10 mins or smth. And this feels not enough for proper climate management using ventilation/conditioning. Too inertial. Need something like 1 per minute or smth.
So I need to get a new devices, but usual specification has no broadcasting frequency information.

**Comment by Ernst79** [comment, marked as answer]
About half of the sensors has a broadcast rate [in the documentation](https://custom-components.github.io/ble_monitor/by_property#temperature), this is displayed as:

![image](https://user-images.githubusercontent.com/24375847/141941507-bc921d7a-2a75-4df0-b132-2176ad8201cb.png)

For most Xiaomi sensors, there is some information about the broadcast rate available. I see you are talking about once per 10 minutes, I therefore assume you have a Xiaomi LYWSD03MMC, is that right? This sensor is special and very popular, because it has the possibility to install custom firmware (ATC firmware) with an online flash tool. This tool will allow you to remove the need for an encryption key, but also to increase the broadcast rate compared to stock Xiaomi firmware. I thought the default value is 1 advertisement per 10 seconds. This would safe you the money for buying new sensors, and flashing isn't very complicated, as it is done in the browser. And it is supported by BLE monitor as well. 

https://github.com/pvvx/ATC_MiThermometer

**Reply by hurricup** [reply]
Yes, you are right. I will try this. ~I'm a bit scared with this firmware update, can it lead to making it inoperable?~ (found the doc how to unbrick it)
I've killed my motherboard once with incorrect bios update :)

Documentation confirms:
> Broadcast Rate 1/10min. (battery level ~1/hr.)*

Thank you!

**Reply by hurricup** [reply]
btw, there are two different firmwares in the doc, do you have experience with them? What is the difference? Another one is https://github.com/atc1441/ATC_MiThermometer

**Comment by Ernst79** [comment]
I would recommend using https://github.com/pvvx/ATC_MiThermometer. If you look at your link, you can also see that this guy is referring to the other:

> Victor @pvvx did some very nice refinings of this custom firmare so i really suggest on cheking it out and even use his version as it offers many more functions including non-volatile storage and a better low power management https://github.com/pvvx/ATC_MiThermometer

I flashed my own sensors without any issue. It's all done over the air via the browser. The main difficulty is getting your computer/browser to connect via BT to your sensor, but if you follow the instructions, you should be fine. 

A little note for you to understand. Clicking on Do Activation will create a new encryption key. You can always flash back the original firmware in the same webpage, but the encryption key will be changes (you can see it in the webtool).
